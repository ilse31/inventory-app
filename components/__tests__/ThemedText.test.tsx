import * as React from "react";
import renderer from "react-test-renderer";

// Hanya mock useColorScheme, bukan seluruh react-native
import * as ReactNative from "react-native";
jest.spyOn(ReactNative, "useColorScheme").mockReturnValue("light");

import { ThemedText } from "../ThemedText";

describe("ThemedText", () => {
  it("renders correctly with default props", () => {
    const tree = renderer
      .create(<ThemedText>Snapshot test!</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with type="title"', () => {
    const tree = renderer
      .create(<ThemedText type='title'>Title</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom lightColor", () => {
    const tree = renderer
      .create(<ThemedText lightColor='#ff0000'>Red Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with style prop", () => {
    const tree = renderer
      .create(<ThemedText style={{ fontSize: 30 }}>Styled Text</ThemedText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
