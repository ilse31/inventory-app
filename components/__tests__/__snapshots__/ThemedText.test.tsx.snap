// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText renders correctly with default props 1`] = `
<Text
  style={
    [
      {
        "color": "#333333",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`ThemedText renders with custom lightColor 1`] = `
<Text
  style={
    [
      {
        "color": "#ff0000",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Red Text
</Text>
`;

exports[`ThemedText renders with style prop 1`] = `
<Text
  style={
    [
      {
        "color": "#333333",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "fontSize": 30,
      },
    ]
  }
>
  Styled Text
</Text>
`;

exports[`ThemedText renders with type="title" 1`] = `
<Text
  style={
    [
      {
        "color": "#333333",
      },
      undefined,
      {
        "fontSize": 32,
        "fontWeight": "bold",
        "lineHeight": 32,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Title
</Text>
`;
